Homework5/src/ro/siit/homework5/Exercise2.java

```diff
@@ -0,0 +1,42 @@
+package ro.siit.homework5;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Exercise2 {
+    public static void main(String[] args) {
+
+        Adress adresa1 = new Adress("Brasov", "Romania");
+        Adress adresa2 = new Adress("Constanta", "Romania");
+        Adress adresa3 = new Adress("Cluj", "Romania");
+        Adress adresa4 = new Adress("Bavaria", "Germany");
+        Adress adresa5 = new Adress("Wien", "Austria");
+        Adress adresa6 = new Adress("Crete", "Greece");
+
+        Hobby ski = new Hobby("ski", 2);
+        ski.addAdress(adresa1);
+        ski.addAdress(adresa4);
+        ski.addAdress(adresa5);
+
+        Hobby swimming = new Hobby("swimming", 5);
+        swimming.addAdress(adresa2);
+        swimming.addAdress(adresa6);
+
+        System.out.println(ski);
+        System.out.println(swimming);
+
+        Person person1 = new Person("Paul", 28);
+        Person person2 = new Person("Oana", 26);
+        Map<Person, Hobby> map = new HashMap();
+        map.put(person1, swimming);
+        map.put(person2, ski);
+        System.out.println(map);
+
+        for (Person p : map.keySet()) {
```

Comment:
If Person would have overridden the equals() and hashCode() methods we would have been able to just call the get() method to retrieve the list of hobbies for "person1". Actually, without implementing these methods, the Person class can't be used as key in a map.